iospi/iotools/cryo_dataset.py

```diff
@@ -0,0 +1,179 @@
+
+    Parameters
+    ----------
+    path : string
+        Path (myfile.h5 or myfile.npy).
+    size : int
+        length of the image side.
+    is_3d : boolean
+        If 2d or 3d.
+
+    Returns
+    -------
+    dataset: torch,
+        greyscale images.
+    """
+    if not os.path.exists(path):
+        raise OSError
+    if path.lower().endswith(".h5"):
+        data_dict = h5py.File(path, "r")
+        all_datasets = data_dict["particles"][:]
+    else:
+        all_datasets = np.load(path)
+    dataset = np.asarray(all_datasets)
+    img_shape = dataset.shape
+    n_imgs = img_shape[0]
+    new_dataset = []
+    s = size
+    if is_3d:
+        dataset = torch.Tensor(dataset)
+        dataset = normalize_torch(dataset)
+        dataset = dataset.reshape((len(dataset),) + (1,) + img_shape[1:])
+    else:
+        if len(img_shape) == 3:
+            for i in range(n_imgs):
+                new_dataset.append(
+                    np.asarray(Image.fromarray(dataset[i]).resize([s, s]))
+                )
+        elif img_shape.ndim == 4:
+            for i in range(n_imgs):
+                new_dataset.append(
+                    np.asarray(Image.fromarray(dataset[i][0]).resize([s, s]))
+                )
+        dataset = torch.Tensor(new_dataset)
+        dataset = normalize_torch(dataset)
+        if len(img_shape) != 4:
+            dataset = dataset.reshape((img_shape[0], 1, s, s))
+    return dataset
+
+
+def normalize_torch(dataset, scale="linear"):
+    """Normalize a tensor.
+
+    Parameters
+    ----------
+    dataset : torch tensor
+        Images.
+    scale : string
+        Methods of normalization
+
+    Returns
+    -------
+    dataset : torch tensor
+        Normalized images.
+    """
+    if scale == "linear":
+        for i, data in enumerate(dataset):
+            min_data = torch.min(data)
+            max_data = torch.max(data)
+            if max_data == min_data:
+                raise ZeroDivisionError
+            dataset[i] = (data - min_data) / (max_data - min_data)
+    return dataset
+
+
+def split_dataset(dataset, batch_size, frac_val):
+    """Separate data in train and validation sets.
+
+    Parameters
+    ----------
+    dataset : torch tensor
+        Images.
+    batch_size : int
+        Batch_size.
+    frac_val : float
+        Ratio between validation and training datasets.
+
+    Returns
+    -------
+    trainset : tensor
+        Training images.
+    testset : tensor
+        Test images.
+    trainloader : tensor
+        Ready to be used by the NN for training images.
+    testloader : tensor
+        Ready to be used by the NN for test images.
+    """
+    n_imgs = len(dataset)
+    n_val = int(n_imgs * frac_val)
+    trainset, testset = random_split(dataset, [n_imgs - n_val, n_val])
+
+    tr = DataLoader(trainset, batch_size=batch_size, shuffle=True, **KWARGS)
+    te = DataLoader(testset, batch_size=batch_size, shuffle=False, **KWARGS)
+    trainloader, testloader = tr, te
```

Comment:
NIT: feels weird to introduce temporary variables just to have things fit in one line. I would prefer :
```
trainloader = DataLoader(trainset, batch_size=batch_size, shuffle=True, **KWARGS)
testloader = DataLoader(testset, batch_size=batch_size, shuffle=False, **KWARGS)
```